### PR TITLE
Java 3733 scaffold for gradle plugin

### PR DESCRIPTION
**Pull request by BrianPhillips2020**
Scaffolding for the new gradle plugin. Registers simple tasks, one hello world and one no-op (partly for my person edification on how task registration works). There are a few things to finish/clear up before this is merged

- [x] Project structure: The project is currently structured like this
```
contrast-sdk-java
|
├── gradle-plugin
|     ├──   build.gradle
|     ├──  settings.gradle
|     └── src
|           └── ...
|
├── src
|   └── ...
├── mvnw
└── pom.xml
```
Should it instead be structured so the `gradle-plugin` and the mvn build for the sdk are in completely separate projects? such as:

```
contrast-sdk-java
|
├── gradle-plugin
|     ├──   build.gradle
|     ├──  settings.gradle
|     └── src
|           └── ...
├── sdk
      ├── src
      |  └── ...
      ├── mvnw
      └── pom.xml
```

- [x] Need to implement branch protections so code cannot be pushed directly to main. Enforced through github? I need to check this out

- [x] Decide how the build pipeline should work. I think separate both the mvn build and the gradle build into different actions have a single workflow relying on both?

### CODE REVIEW

File: gradle-plugin/build.gradle
Comment (stiemannkj1) [inline gradle-plugin/build.gradle:1]: Are people still building on Java 8? I think we should set the compiler release level to 8. Or at least 11.

File: .github/workflows/build-gradle-plugin.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - main
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```
Comment (stiemannkj1) [inline .github/workflows/build-gradle-plugin.yml:16]: You need to use the full has here to be safe from supply chain attacks.

File: .github/workflows/build-gradle-plugin.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        run: echo "The ${{ github.repository }} repo has been cloned to the runner
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
```
Comment (stiemannkj1) [inline .github/workflows/build-gradle-plugin.yml:20]: You need to use the full has here to be safe from supply chain attacks.

File: .github/workflows/build-gradle-plugin.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Setup Java
+        uses: actions/setup-java@v2
```
Comment (stiemannkj1) [inline .github/workflows/build-gradle-plugin.yml:23]: You need to use the full has here to be safe from supply chain attacks.

File: gradle-plugin/src/test/java/com/contrastsecurity/gradle/plugin/RegistrationTests.java
Comment (stiemannkj1) [inline gradle-plugin/src/test/java/com/contrastsecurity/gradle/plugin/RegistrationTests.java:1]: Is there really any value in these tests if we have integration tests? I guess keep them for now, but I think we might end up deleting them. I'd rather validate with the tests that actually run stuff.

**Review by stiemannkj1 — APPROVED** [review]

**Comment by stiemannkj1** [comment]
Nice work!

**Comment by stiemannkj1** [comment]
> Should it instead be structured so the gradle-plugin and the mvn build for the sdk are in completely separate projects? 

Yes, please make that change too.

File: gradle-plugin/src/test/java/com/contrastsecurity/gradle/plugin/RegistrationTests.java
Comment (BrianPhillips2020) [reply gradle-plugin/src/test/java/com/contrastsecurity/gradle/plugin/RegistrationTests.java:1]: We'll probably end up deleting them. This is the most basic smoke test you can make for a gradle plugin. Everything I've seen in researching this has this little test just verifying they're registered. Might be worth having just for the very very simple case of identifying if registration breaks, but I'm not attached

File: gradle-plugin/build.gradle [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+group = 'com.contrastsecurity.gradle.plugin'
+
+repositories {
+    mavenCentral()
+}
+
+gradlePlugin {
+    plugins {
+        java {
+            id = 'com.contrastsecurity.java'
```
Comment (rszewczyk) [inline gradle-plugin/build.gradle:17]: How are we choosing the plugin name?

File: gradle-plugin/build.gradle
Comment (rszewczyk) [reply gradle-plugin/build.gradle:1]: @stiemannkj1 are you saying we _should_ support building as far back as Java 8 or 11? Why though?

**Review by rszewczyk — APPROVED** [review]

**Review by rszewczyk — APPROVED** [review]

File: gradle-plugin/build.gradle
Comment (rszewczyk) [reply gradle-plugin/build.gradle:1]: 🤔 It looks like Gradle [still supports Java 8](https://docs.gradle.org/current/userguide/compatibility.html), so long as you're not using the daemon, even on it's latest versions, so I guess that is prudent.

File: gradle-plugin/build.gradle (hunk 4 above)
Comment (BrianPhillips2020) [reply gradle-plugin/build.gradle:17]: I forgot to bring this up, no idea 🙃 nobody picked a name yet, this is completely arbitrary. Easily changeable at least, I might make permanently choosing that a requirement for before we actually release

Comment (rszewczyk) [reply gradle-plugin/build.gradle:17]: Yeah, it's not something we need to decide until we publish. Calling the plugin "java" seems a bit vague, but I can't think of something better at the moment. 🤷

**Comment by stiemannkj1** [comment]
Added branch protection for `main` myself since I don't think you have permissions.

**Pull request merged**